updates for spring 2022 (#489)
AaltoSciComp/scicomp-docs: training/scip/summer-kickstart-2022.rst

=====================================================================
June 2022 / Intro to Scientific Computing (FGCI HPC Summer Kickstart)
=====================================================================

.. warning::

>    This page is still under construction. The dates for June 2022 introduction to HPC
>    will be Tue 6, Wed 7, Thu 8 June 2022. Below you can read last year's page.

Reviewer: ```suggestion
   This courses schedule is still under development, below you see the template from last year.
   The dates will be 7-9 June 2022 (T-Th).
   If you have requests for the schedule or would like to offer a talk as part of this course, please 
   get in touch with us.
```

* **Livestream**: anyone may watch at https://twitch.tv/coderefinery,
  no registration needed!

* **Zoom**: if you register, you will be able to attend a Zoom meeting
  that includes interactive breakout rooms and hands-on help.  We
  watch the livestream for the main material.

* **HackMD**: instead of chat, this is used for Q&A.  See the
  `CodeRefinery HackMD manual
  <https://coderefinery.github.io/manuals/hackmd-mechanics/>`__ for
  how this works.

Schedule
--------

The daily schedule will be adjusted based on the
audience; below is the tentative plan.  There will be frequent
breaks. You will be given time to try and ask, it’s more like an
informal help session to get you started with the computing
resources.  All times are EEST (Helsinki) time.

* **Preparatory material**.  Each year the first day has varying
  topics presented.  We don't repeat these every year, but we strongly
  recommend that you watch these videos yourself as preparation:

  * `Basic Linux shell and scripting
    <https://www.youtube.com/watch?v=ESXLbtaxpdI&list=PLZLVmS9rf3nN_tMPgqoUQac9bTjZw8JYc&index=3>`__
  * `Scientific computing workflows
    <https://www.youtube.com/watch?v=ExFbc5EikU0>`__

> * **Day #1 (Mon 7.jun):** Basics and background

Reviewer: ```suggestion
* **Day #1 (Tue 7.jun):** Basics and background
```

  * 12:50: Break

  * 13:00: **Currently available resources at CSC** CSC is the Finnish
    center for scientific computing, and also has many resources for
    research. (`Slides <https://kannu.csc.fi/s/3K8q93XSwtSgHEa>`__)
    (Jussi Enkovaara, CSC).

  * 13:45: Break

  * 14:00: **Git intro**: why you need version control for any
    scientific work and how to get started.  We don't go in depth into
    theory, but talk about the simplest usage by yourself. (Richard
    Darst, Jarno Rantaharju)

  * 14:45: Break

  * 15:00: **Your future career in scientific computing (and this
    course).** (:doc:`Notes <summer-kickstart/future>`) (Enrico Glerean, TBA)

  * 15:15: **Connecting to the cluster**, hands-on.
    Get connected in preparation for day 2 (Enrico Glerean)

    * Aalto: :doc:`Connecting to Triton tutorial
      </triton/tut/connecting>` – if you can ssh to Triton and run
      ``hostname``, you are ready for tomorrow.
    * Helsinki: `general information <https://wiki.helsinki.fi/display/it4sci/HPC+SUMMER+KICKSTART+2021>`__
    * Tampere: `Connecting to Narvi <https://narvi-docs.readthedocs.io/narvi/tut/connecting.html>`__

> * **Day #2 (Tue 8.jun):** Basic use of a cluster (Richard Darst, Simo

Reviewer: ```suggestion
* **Day #2 (Wed 8.jun):** Basic use of a cluster (Richard Darst, Simo
```

    * 13:50: Break

    * 14:00: Short talk: `Radovan Bast <https://bast.fr/>`__ (UiT The Arctic University of Norway): `Asking for help with supercomputers <https://cicero.xyz/v3/remark/0.14.0/github.com/bast/help-with-supercomputers/main/talk.md/#1>`__ 

      * How should you write support requests so that you get quick 
        (and useful!) answers? Radovan, one of the founders of 
        `CodeRefinery <https://coderefinery.org/>`__, will talk about how we can all improve 
        the dialogue between supercomputer user community and support staff 
        so that we always remain respectful and try to learn and solve problems together.
       

    * 14:35: :doc:`/triton/tut/interactive`

      * The basic Slurm concepts are the same across all clusters (at
	least all those that use Slurm, but that is everyone in
	Finland).  However, partition names may be different.  You can
	list partitions at your site using ``sinfo -O partition`` and
	list nodes at your site with ``sinfo -N``.  How these work
	will vary depending on your site - definitely read up on this.

    * 14:50: Break

    * 15:00 Continuing with interactive slurm jobs and exercises

    * 16:00: End

> * **Day #3 (Wed 9.jun):** Advanced cluster use (Simo Tuomisto, Richard

Reviewer: ```suggestion
* **Day #3 (Thu 9.jun):** Advanced cluster use (Simo Tuomisto, Richard
```
